### PR TITLE
DG-17834: svg support

### PR DESCRIPTION
**Pull request by richardadjogah**
This branch improves SVG support by changing the default image background color to white as well as upgrading the `sharp` version. SVG background color can be manually set with the background  command.

### CODE REVIEW

File: src/processImage.js [hunk 1]
```diff
@@ -81,16 +80,20 @@ exports.processImage = function(req, res) {
...
 			function(finalImage, metadata, processingStartTime, next) {
+				var finalSharpObject = exports.processFlags(sharp(finalImage.body),
+					req.logger, req.commandsString, metadata);
 
-				// Use defaults for file format/quality. Convert unknown image types (such as gifs) to jpeg
-				var finalSharpObject = sharp(finalImage.body)
-					.quality(QUALITY)
-					.compressionLevel(COMPRESSION_LEVEL)
-					.toFormat(req.imageExtension);
+				// Default PNG/SVG backgrounds to white instead of black
```
Comment (erindethorne) [inline src/processImage.js]: Don't pngs support transparant backgrounds? Would that be a possibility here instead of white?

Comment (richardadjogah) [inline src/processImage.js]: @erindethorne good idea.

File: src/processImage.js [hunk 2]
```diff
@@ -63,8 +63,7 @@ exports.processImage = function(req, res) {
 					req.logger.debug('Converting gif to jpeg format');
 					sharp(body)
 						.toFormat('jpeg')
-						.quality(100)
-						.toBuffer()
+						.toBuffer({quality: QUALITY})
```
Comment (erindethorne) [inline src/processImage.js]: Looks like we used to do `quality: 100` and with this change it is reduced to `quality: 70`. What's the background for this change? Could it be a visible difference to our users?

Comment (richardadjogah) [inline src/processImage.js]: This was an accident, but I went ahead and made gifs convert to PNG rather than JPG to avoid this issue entirely.

File: src/processImage.js [hunk 3]
```diff
@@ -58,12 +58,11 @@ exports.processImage = function(req, res) {
 
 				// Convert gifs to jpeg before any operations
 				if (req.imageExtension === 'gif') {
-					req.imageExtension = 'jpeg';
+					req.imageExtension = 'png';
```
Comment (erindethorne) [inline src/processImage.js]: Could there be any user-facing difference from converting gifs to pngs instead of jpegs?

Probably because of my unfamiliarity with the codebase, but it's also a little unclear to me why this change was made and how it's relevant to handling svgs.

Comment (richardadjogah) [inline src/processImage.js]: It would be a straight upgrade in all scenarios because pngs are lossless. This change isn't directly related to SVG handling but is a nice QoL update that should have been there to begin with.

Comment (erindethorne) [inline src/processImage.js]: Sorry now I'm just super curious - why did we use jpgs to begin with if pngs are give a better outcome?

Comment (richardadjogah) [inline src/processImage.js]: Not sure actually. It might have just been a holdover from the old app.

Comment (erindethorne) [inline src/processImage.js]: Seems like it could be worth finding out, if there's anybody who might know. Just to verify whether this is a good change to make, since it's unrelated to the ticket this addresses.

File: src/processImage.js [hunk 4]
```diff
@@ -81,16 +80,17 @@ exports.processImage = function(req, res) {
...
-					.toFormat(req.imageExtension);
-
-				finalSharpObject = exports.processFlags(finalSharpObject, req.logger, req.commandsString, metadata);
+				// Don't convert SVGs into PNGs until now so that fills work in the future
+				if ((req.imageExtension === 'svg' || req.imageExtension === 'png') && req.commandsString.indexOf('/background/') === -1) {
+					req.imageExtension = 'png';
+				}
 
-				finalSharpObject.toBuffer()
+				// Use defaults for file format/quality. Convert unknown image types (such as gifs) to jpeg
```
Comment (erindethorne) [inline src/processImage.js]: Is this comment still true? Are we converting unknown image types to svgs now?

**Comment by erindethorne** [comment]
Looks like the build for this is also failing: http://websbuild201.vistaprint.net:8085/browse/NIP-APP45-5

File: src/processImage.js (hunk 3 above)
Comment (richardadjogah) [inline src/processImage.js]: The only read consideration here is the filesize/compression ratio tradeoff, but given that gifs tend to be low res due to having animation frames and generally being a terrible outdated file format, the absolute size isn't something I would worry about. I'm totally confident in this being a direct upgrade in image quality for gifs with little to no negative effects. I also don't think anyone who worked on the old image processor app is still at the company.

**Comment by richardadjogah** [comment]
@erindethorne The build is failing because apparently the host we've been using to download vips  (http://www.vips.ecs.soton.ac.uk/supported/) is out of date and cant serve the source code anymore. Given that the sharp installation instructions (http://sharp.dimens.io/en/stable/install/) state that there's a binary that now comes pre-installed with the npm dependency, we shouldn't have to compile vips from source at all in Docker anymore. This way has never been tested on our boxes, but I see no reason why it shouldn't work given that the Docker machine is based off Debian. Just something to note though.

File: src/commands.js [hunk 5]
```diff
@@ -86,18 +86,19 @@ var height = function(sharpobject, params, metadata) {
 var jpeg = function(sharpobject, params, metadata) {
 	// check for optional argument
 	if (params[1]) {
-		return sharpobject.background('#FFFFFF').flatten().jpeg({quality: parseInt(params[1].substring(1))});
-	} else {
-		return sharpobject.background('#FFFFFF').flatten().jpeg();
+		metadata.flags.quality = parseInt(params[1].substring(1));
```
Comment (nchase) [inline src/commands.js]: just wondering: is this function meant to only mutate `metadata`, which is then used in other functions, or is `metadata` meant to be used somehow in this function?

Comment (richardadjogah) [inline src/commands.js]: That line specifically mutates metadata if the option `quality` argument is passed in, but this command always modifies the images regardless.
